src/toil/jobStores/abstractJobStore.py

```diff
@@ -964,12 +983,29 @@ def readFile(self, jobStoreFileID, localFilePath, symlink=False):
 
     @abstractmethod
     @contextmanager
-    def readFileStream(self, jobStoreFileID):
+    def readFileStream(self, jobStoreFileID, mode='b', encoding=None, errors=None):
```

Comment:
Same here as above.  Remove `mode` and only use `encoding`.  The default of `None` means no encoding, and just `bytes` will be used.